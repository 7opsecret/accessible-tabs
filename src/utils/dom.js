export const setAttributes = (element, attributes) => {
    for(let key in attributes) {
      element.setAttribute(key, attributes[key]);
    }
}