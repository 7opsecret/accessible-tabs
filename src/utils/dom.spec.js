import { setAttributes } from './dom';

describe('Utils: dom', () => {
    describe('#setAttributes', () => {
        it('should set attributes to element correctly', () => {
            const mockEl = document.createElement('div');

            setAttributes(
                mockEl,
                {
                    'aria-labelledby': 'fake-id',
                    role: 'tabpanel',
                    tabindex: '0'
                }
            );

            expect(mockEl.outerHTML).toMatchSnapshot();
        });
    });
});