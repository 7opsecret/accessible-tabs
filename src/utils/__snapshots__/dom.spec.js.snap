// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Utils: dom #setAttributes should set attributes to element correctly 1`] = `"<div aria-labelledby=\\"fake-id\\" role=\\"tabpanel\\" tabindex=\\"0\\"></div>"`;
